sdk/keyvault/keyvault-certificates/samples-dev/helloWorld.ts

```diff
@@ -23,47 +25,60 @@ export async function main(): Promise<void> {
 
   const client = new CertificateClient(url, credential);
 
+  // Create unique certificate name
   const uniqueString = new Date().getTime();
   const certificateName = `cert${uniqueString}`;
 
-  // Creating a self-signed certificate
+  // Create a self-signed certificate
   const createPoller = await client.beginCreateCertificate(
     certificateName,
     DefaultCertificatePolicy
   );
 
+  // Wait for long running operation to complete
```

Comment:
This isn't quite accurate - the poller exposes a [`pollUntilDone`](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-lro/src/poller.ts#L59) method which will resolve when the operation is complete, but you can get the latest result of the operation synchronously with [`getResult`](https://github.com/Azure/azure-sdk-for-js/blob/ba54673c1f5e22601969341eeebe5ba6c5363e92/sdk/core/core-lro/src/poller.ts#L89-L95)

@sadasant can confirm but I believe here you are likely getting the pending certificate before it was created https://docs.microsoft.com/en-us/azure/key-vault/certificates/create-certificate#asynchronous-process

Reply:
That is correct! Here’s the source: https://github.com/Azure/azure-sdk-for-js/blob/569b898dce03474c75c612c4d2d3c198cfd0c9dc/sdk/core/core-lro/src/poller.ts#L512-L515

getResult doesn’t wait for anything to happen. In this sample, it will show the first result, which will be a pending certificate.

A fully signed certificate can only be obtained with `pollUntilDone`.